[Multisig] MultisigCredential / Traverse cleanup

c4t-ag opened the pull request:
## Why this should be merged
- RegisterNode SubnetAuth not multisig capable (fixed)
- Introduce MultisigCredential to be able to contol if a Multisig alias should be traversed or not.
- Make return values in Traverse more descriptive by replacing bool with 4-state enum

## How this works
- RegisterNode used non-msig capable Permission verification -> Use Msig version
- Signatures passed in Credentials don't include any information about MultisigAlias.
Assume you have a CG with A1, A2, M, A3 and your signatures should verify the 3 A's, something must be provided to tell the logic that M should be skipped. This is what SigIndices are normally meant for, but again, there are no sigIndices for Multisig Alias.

## How this was tested
- Existing unit tests

evlekht commented on vms/secp256k1fx/camino_credential.go:
```diff
@@ -0,0 +1,69 @@
+
+type CredentialIntf interface {
+	Signatures() [][crypto.SECP256K1RSigLen]byte
+	Verify() error
```
maybe include Verifiable interface instead of Verify func?

c4t-ag replied:
ok

evlekht commented on vms/secp256k1fx/camino_credential.go:
```diff
@@ -0,0 +1,69 @@
+
+type MultisigCredential struct {
+	Credential      `serialize:"true"`
+	MultiSigAliases []ids.ShortID `serialize:"true" json:"multisigAliases"`
```
Is this multisig aliases or just addresses from one multisig?

c4t-ag replied:
Only aliases, a list of aliases which should be checked, see PR description.
There can be wired constellations where verification could fail (theoretically, but possible)
Lets say we have a1, M1{a2, a3 => T2}, a2 => T2 and you provide signatures a1 and a2.
Without knowing that we shouldn't ignore M1, the verification would fail.

peak3d committed:
Multisig Fixes and Cleanup
- Remove old getOwner 1-level multisig function
- Allow Array[Creds] in Unordered verification, use it now also for Validator C-Member check
- Introduce MultiAliasCredential which allows selection of multisigAliases

evlekht commented on vms/platformvm/txs/builder/camino_builder.go:
```diff
@@ -509,23 +509,24 @@ func (b *caminoBuilder) NewRegisterNodeTx(
-
-	consortiumSigners, err := getSigners(keys, consortiumMemberOwner.Addrs)
+	kc := secp256k1fx.NewKeychain(keys...)
+	out, consortiumSigners, err := kc.SpendMultiSig(
```
```suggestion
	in, consortiumSigners, err := kc.SpendMultiSig(
```
